plugins/cpp/service/src/diagram.cpp

```diff
@@ -329,14 +329,11 @@ std::string Diagram::getDetailedClassNodeLabel(const AstNodeInfo& nodeInfo_)
   _cppHandler.getReferences(nodes, nodeInfo_.id,
     CppServiceHandler::DATA_MEMBER, {});
 
-  std::map<core::AstNodeId, std::map<std::string, std::string>> props
-    = getProperties(nodes);
-
   for (auto it = nodes.begin(); it != nodes.end(); ++it)
   {
     std::string visibility = visibilityToHtml(*it);
     std::string content = memberContentToHtml(*it,
-      util::escapeHtml(it->astNodeValue + " : " + props[it->id]["Type"]));
+      util::escapeHtml(it->astNodeValue + " : " + getProperty(it->id, "Type")));
```

Comment:
Good point. As the function signature has been placed in the `astNodeValue` attribute, it was no longer necessary to query properties for multiple functions. That's why I also reverted the changes of `getProperties()` API function. However that's true that now this results as many transactions as many attributes a class has in the diagram. I'll figure something out for this.